### ISSUE TITLE
Update link states to match the design

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyone#110]
The links need to be updated to match the new styles:

![Regular links have a solid underline, hovered links have a dotted underlined, and focused links have a solid box shadow with a border radius](https://user-images.githubusercontent.com/7422055/93844529-b7af3780-fc9d-11ea-90aa-334ed7ed56f5.png)

**Comment by melchoyce** [comment]
Here's a coded mockup of the link styles: https://codepen.io/melchoyce/pen/jOqdNPm

**Comment by fabiankaegy** [comment]
I can pick this one up and create a PR for it

**Comment by aristath** [comment]
The problem with using border & border-radii for links can be seen when the link text flows to the below line. Example from the codepen posted above - I just moved the closing `</a>` so the link flows to the line below.

With borders:

![Screenshot_2020-09-24 TT1 Links 3(1)](https://user-images.githubusercontent.com/588688/94107431-1e6f5500-fe46-11ea-8429-466bb327608b.png)

With `outline: 2px solid; text-decoration: none; outline-offset: 2px;`:

![Screenshot_2020-09-24 TT1 Links 3(2)](https://user-images.githubusercontent.com/588688/94107776-d270e000-fe46-11ea-94b2-e78a8525a650.png)

**Comment by fabiankaegy** [comment]
@aristath I have created a PR before seeing your notes but can happily update everything once a decision regarding the final styling was made.

**Comment by melchoyce** [comment]
I think the more I look at it, the less I'm tied to the border-radius. Let's try this instead:

```
a:focus {
  outline: 2px solid var(--black);
  text-decoration: none;
}

```
Thanks @aristath @fabiankaegy!

**Comment by melchoyce** [comment]
Yesssssssss:

https://twitter.com/argyleink/status/1309172888243712002?s=20

**Comment by carolinan** [comment]
What about the post and page titles? Do we keep them underlined by default?

**Comment by melchoyce** [comment]
Would love a11y confirmation here.
